lambdas/transcode/index.py

```diff
@@ -49,44 +62,56 @@ def lambda_handler(request):
     Generate previews for videos in S3
     """
     url = request.args['url']
-    format = request.args.get('format', 'mp4')
-    width_str = request.args.get('width', '320')
-    height_str = request.args.get('height', '240')
-    duration_str = request.args.get('duration', '5')
+    format = request.args['format']
 
-    try:
-        width = int(width_str)
-        if not MIN_WIDTH <= width <= MAX_WIDTH:
-            raise ValueError
-    except ValueError:
-        return make_json_response(400, {'error': f"Invalid 'width'; must be between {MIN_WIDTH} and {MAX_WIDTH}"})
+    def _parse_param(name, default_value, min_value, max_value):
+        value_str = request.args.get(name)
+        if value_str is None:
+            return default_value
 
-    try:
-        height = int(height_str)
-        if not MIN_HEIGHT <= height <= MAX_HEIGHT:
-            raise ValueError
-    except ValueError:
-        return make_json_response(400, {'error': f"Invalid 'height'; must be between {MIN_HEIGHT} and {MAX_HEIGHT}"})
+        try:
+            value = type(default_value)(value_str)
+            if not min_value <= value <= max_value:
+                raise ValueError
+            return value
+        except ValueError:
+            raise ValueError(f"Invalid {name!r}; must be between {min_value} and {max_value}")
 
     try:
-        duration = float(duration_str)
-        if not 0 < duration <= 10:
-            raise ValueError
-    except ValueError:
-        return make_json_response(400, {'error': "Invalid 'duration'"})
+        width = _parse_param('width', 320, 10, 640)
+        height = _parse_param('height', 240, 10, 480)
+        duration = _parse_param('duration', 5.0, 0.1, 10)
+        audio_bitrate = _parse_param('audio_bitrate', 128, 64, 320)
+        file_size = _parse_param('file_size', MAX_FILE_SIZE, 1024, MAX_FILE_SIZE)
+    except ValueError as ex:
+        return make_json_response(400, {'error': str(ex)})
+
+    format = LEGACY_FORMATS.get(format, format)
+    category = format.split('/')[0]
+
+    format_params = []
+    if category == 'audio':
+        format_params.extend([
+            '-b:a', f'{audio_bitrate}k',
+            '-vn',  # Drop the video stream
+        ])
+    elif category == 'video':
+        format_params.extend([
+            "-vf", ','.join([
+                f"scale=w={width}:h={height}:force_original_aspect_ratio=decrease",
+                "crop='iw-mod(iw\\,2)':'ih-mod(ih\\,2)'",
+            ]),
+        ])
```

Comment:
Do you want `else throw` here?

Reply:
LOL. I used to have `else: assert False`, and got a pycov error cause that line wasn't covered. But, obviously, it can't be covered since it can't happen. What's the solution here?

Reply:
You _could_ patch `FORMATS`... up to you though